.github/workflows/testing.yml

```diff
@@ -44,12 +45,50 @@ jobs:
             golang: 1.18
             coveralls: false
 
+    env:
+      TNT_BUILD_INSTALL_PATH: /home/runner/tnt-install
+
     steps:
       - name: Clone the connector
         uses: actions/checkout@v2
 
+      - name: Cache Tarantool build
+        if: matrix.tarantool == '1.6.9'
+        id: cache-tnt-build
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.TNT_BUILD_INSTALL_PATH }}
+          key: cache-tnt-${{ matrix.tarantool }}
+      - name: Clone Tarantool ${{ matrix.tarantool }}
+        if: matrix.tarantool == '1.6.9' && steps.cache-tnt-build.outputs.cache-hit != 'true'
+        uses: actions/checkout@v3
+        with:
+          repository: tarantool/tarantool
+          ref: ${{ matrix.tarantool }}
+          path: tarantool
+          fetch-depth: 0
+          submodules: true
+
+      - name: Build Tarantool ${{ matrix.tarantool }}
+        if: matrix.tarantool == '1.6.9' && steps.cache-tnt-build.outputs.cache-hit != 'true'
+        run: |
+          sudo apt-get -y install git build-essential cmake make zlib1g-dev \
+            libreadline-dev libncurses5-dev libssl-dev \
+            libunwind-dev libicu-dev python3 python3-yaml \
+            python3-six python3-gevent
+          cd ${GITHUB_WORKSPACE}/tarantool
+          mkdir build && cd build
+          cmake .. -DCMAKE_BUILD_TYPE=Release
+          make
+          make DESTDIR=${TNT_BUILD_INSTALL_PATH} install
```

Comment:
Wow. I would do the following instead:

* Build the package once for Ubuntu Focal.
* Agree with @LeonidVas and @artembo where to place such special packages.
* Optionally add support for setup-tarantool.

I don't insist: if it works, it is likely okay. However we can do it a bit better (faster, more durable, less code).